torch/distributed/_sharded_tensor/ops/linear.py

```diff
@@ -138,28 +175,40 @@ def _handle_col_wise_sharding(input, world_size, weight, local_shard_t, bias, pg
         input: matrix to be multiplied with the sharded weight.
         world_size: number of ranks.
         weight: shareded weight tensor.
+        rank: # of cuda process.
         local_shard_t: row-wise shared local weight used for lookup.
         bias: bias term of linear op.
         pg: process group.
 
-    Returns: final result of linear operation.
-    """
-    return (
-        _handle_col_wise_sharding_base(
-            torch.matmul,
-            weight.size(0),
-            len(input.size()) - 1,
-            input,
-            world_size,
-            weight,
-            local_shard_t,
-            pg,
-        )
-        + bias
+    Returns:
+        A :class:`ShardedTensor` object which filled with local intermediate results.
+    """
+    # allgather the inputs first.
+    gathered_inputs = all_gather(input, group=pg)
+    (start_pos, chunk_size) = get_chunk_sharding_params(
+        bias.size(0), world_size, weight._sharding_spec, rank
+    )
+    local_bias = _BiasTensorNarrow.apply(
+        world_size, start_pos, chunk_size, weight, pg, bias
+    )
+    results = [torch.tensor(1)] * world_size
```

Comment:
Why can't this just be `None`?

Reply:
Will make it None and add a mypy suppression.